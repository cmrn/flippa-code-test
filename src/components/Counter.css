.Counter {
  padding: 10px 0;
  display: flex;
  border-bottom: 1px solid #ccc;
  align-items: center;
  font-size: 20px;
}

.Counter:first-of-type {
  border-top: 1px solid #ccc;
}

.Counter-title {
  flex-grow: 1;
  padding: 0 10px;
}

.Counter-count {
  padding: 0 10px;
}

.Counter-button {
  border: none;
  background: none;
}

.Counter-delete {
  font-size: 15px;
}