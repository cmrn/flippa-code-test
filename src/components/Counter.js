import React from 'react';
import PropTypes from 'prop-types';
import './Counter.css';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faMinus from '@fortawesome/fontawesome-free-solid/faMinus';
import faPlus from '@fortawesome/fontawesome-free-solid/faPlus';
import faTimes from '@fortawesome/fontawesome-free-solid/faTimes';

const Counter = ({ title, count }) => (
  <div className="Counter">
    <button className="Counter-button Counter-delete"><FontAwesomeIcon color="crimson" icon={faTimes} /></button>
    <span className="Counter-title">{title}</span>
    <button className="Counter-button"><FontAwesomeIcon icon={faMinus} /></button>
    <span className="Counter-count">{count}</span>
    <button className="Counter-button"><FontAwesomeIcon icon={faPlus} /></button>
  </div>
);

Counter.propTypes = {
  title: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
};

export default Counter;
