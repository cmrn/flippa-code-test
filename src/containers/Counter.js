import React from 'react';
import Counter from '../components/Counter';
import { view } from 'react-easy-state';
import counterStore from '../counterStore';

const CounterContainer = ({ id, ...props }) => {
  const counter = counterStore.counters[id];
  if(!counter) return null;
  return (
    <Counter
      {...props}
      title={counter.title}
      count={counter.count}
      onIncrement={() => counterStore.increment(counter.id)}
      onDecrement={() => counterStore.decrement(counter.id)}
      onDelete={() => counterStore.delete(counter.id)}
    />
  );
};

export default view(CounterContainer);
