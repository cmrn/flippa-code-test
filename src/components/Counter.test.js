import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Counter from './Counter';

it('renders the counter with title, count, add, subtract, and remove buttons', () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<Counter title="A title" count={161} />);
  expect(tree).toMatchSnapshot();
});
