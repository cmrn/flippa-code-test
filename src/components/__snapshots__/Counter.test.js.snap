// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the counter with title, count, add, subtract, and remove buttons 1`] = `
<div
  className="Counter"
>
  <button
    className="Counter-button Counter-delete"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="crimson"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </button>
  <span
    className="Counter-title"
  >
    A title
  </span>
  <button
    className="Counter-button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f068",
            "M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
          ],
          "iconName": "minus",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </button>
  <span
    className="Counter-count"
  >
    161
  </span>
  <button
    className="Counter-button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f067",
            "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
          ],
          "iconName": "plus",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </button>
</div>
`;
